components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/execution/PreIssueAccessTokenRequestBuilder.java

```diff
@@ -199,11 +264,20 @@ private AccessToken getAccessToken(OAuthTokenReqMessageContext
     }
 
     private OAuthAppDO getAppInformation(OAuthTokenReqMessageContext tokenMessageContext)
-            throws InvalidOAuthClientException, IdentityOAuth2Exception {
+            throws ActionExecutionRequestBuilderException {
 
-        return OAuth2Util.getAppInformationByClientId(
-                tokenMessageContext.getOauth2AccessTokenReqDTO().getClientId(),
-                tokenMessageContext.getOauth2AccessTokenReqDTO().getTenantDomain());
+        OAuthAppDO oAuthAppDO;
+        try {
+            oAuthAppDO = OAuth2Util.getAppInformationByClientId(
+                    tokenMessageContext.getOauth2AccessTokenReqDTO().getClientId(),
+                    tokenMessageContext.getOauth2AccessTokenReqDTO().getTenantDomain());
```

Comment:
If tenant domain and client Id is being used in multiple places better to assign to a local variable and use for clarity.

Reply:
I normally practice using getters as much as possible since creating a reference adds a memory hit even smaller and shorter time. If we execute and modify the value, yes I am totally on your opinion too. WDYT ?

Reply:
ok make sense